### PR TITLE
Block toxic comments using the Perspectives API

### PR DESCRIPTION
**Pull request by n-wach**
Waiting on #40

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 1]
```diff
@@ -54,14 +62,22 @@
   private static final String PARAM_TOPIC = "topic";
   private static final String PARAM_CONTENT = "content";
   private static final String PARAM_CURSOR = "cursor";
+  private static final float MAX_ACCEPTABLE_TOXICITY = 0.8f;
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:65]: Can you add a comment explaining what 0.8 represents? If you don't know then you can omit the comment.

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 2]
```diff
@@ -141,4 +163,28 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(threadJson);
   }
+
+  /**
+   * Restures {@code true} if a comment has acceptable levels of {@code TOXICITY}.
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:168]:
```suggestion
   * Returns {@code true} if a comment has acceptable levels of {@code TOXICITY}.
```

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 3]
```diff
@@ -141,4 +163,28 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+   */
+  public boolean isAcceptableComment(AnalyzeCommentResponse analysis) {
+    AttributeScore toxicity = analysis.getAttributeScores().get(AttributeType.TOXICITY);
+    float value = toxicity.getSummaryScore().getValue();
+    return value <= MAX_ACCEPTABLE_TOXICITY;
+  }
+
+
+  /**
+   * Returns an {@AnalyzeCommentResponse} from the Perspective API analysing {@content}.
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:178]:
```suggestion
   * Returns an {@code AnalyzeCommentResponse} from the Perspective API analyzing {@code content}.
```

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java (hunk 1 above)
Comment (n-wach) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:65]: All I know is that it is a measure of toxicity on a scale of 0 to 1

Comment (n-wach) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:65]:
```suggestion
  /** Toxicity is measured on a scale from 0 to 1.  This should block most directed insults and profanity. */
  private static final float MAX_ACCEPTABLE_TOXICITY = 0.8f;
```

File: portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java [hunk 4]
```diff
@@ -54,14 +62,23 @@
   private static final String PARAM_TOPIC = "topic";
   private static final String PARAM_CONTENT = "content";
   private static final String PARAM_CURSOR = "cursor";
+  /** Toxicity is measured on a scale from 0 to 1.  This should block most directed insults and profanity. */
```
Comment (n-wach) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java:65]:
```suggestion
  /* Toxicity is measured on a scale from 0 to 1.  This should block most directed insults and profanity. */
```
Normal comments for fields

**Pull request merged**